migration-guides/0.14-Migration-Guide.md

```diff
@@ -0,0 +1,597 @@
++
++ functionName' :: forall a. a -> a
++ functionName' = functionNameImpl
+```
+
+## Library Changes
+
+Several core library changes accompanied the PureScript 0.14 compiler release. This guide covers notable changes to the `purescript` (core), `purescript-contrib`, `purescript-node`, and `purescript-web` organizations.
+
+However, the changes listed below are not exhaustive. As of the PureScript 0.14 release, all libraries in these organizations contain a CHANGELOG.md file which lists any breaking changes in their latest release. You should consult the changelogs for any libraries you depend on in addition to this migration guide.
+
+### Library Restructuring
+
+Several core libraries have been restructured. In almost all cases functionality has been preserved, but some functions, types, and / or instances have been moved to new locations.
+
+#### Monomorphic proxies and the `proxy` library have been deprecated, with `Proxy` merged into `prelude`.
+
+PureScript 0.14 introduced support for polymorphic kinds, which means we no longer need a proliferation of different proxy types which differ only in the kind of their parameter. For example, we previously had many proxy types:
+
+```purs
+data Proxy (a :: Type) = Proxy
+data Proxy2 (a :: Type -> Type) = Proxy2
+data Proxy3 (a :: Type -> Type -> Type) = Proxy3
+data SProxy (a :: Symbol) = SProxy
+data RProxy (row :: # Type) = RProxy
+data RLProxy (row :: RowList) = RLProxy
+-- ...and so on.
+```
+
+All proxy types except for `Proxy` have been deprecated, and `Proxy` is now implemented with a polymorphic kind:
+
+```purs
+-- | Proxy type for all kinds.
+data Proxy :: forall k. k -> Type
+data Proxy a = Proxy
+```
+
+In addition, the `proxy` library has been deprecated and all relevant functionality moved into `prelude`.
+
+**To fix:**
+
+You should remove `proxy` as a dependency of your library or application, as the relevant code now lives in the `prelude` package.
+
+You do not need to update code using proxy types like `Proxy2` and `SProxy`, because these types have been deprecated but will not be removed until the PureScript 0.15 compiler release. However, if you'd like to get ahead of this change, you should replace all uses of proxy types other than `Proxy` to use `Proxy` instead. For example:
+
+```purs
+-- Code written pre-PureScript 0.14, which will compile with PureScript 0.14
+-- but will not compile with PureScript 0.15.
+getFoo = Record.get (SProxy :: SProxy "foo")
+
+-- Code written for PureScript 0.14 which will continue to work with
+-- PureScript 0.15.
+getFoo = Record.get (Proxy :: Proxy "foo")
+```
+
+
+#### The `generics-rep` library has been merged into `prelude`
+
+`Generic` has been moved to the bottom of the core ecosystem because it's widely-used class with compiler support, but it didn't previously cover some common types like `Tuple` or `Either`. This change enables all of the data types across the core libraries which have a sensible `Generic` instance to have one.
```

Comment:
It's not really accurate to say this change enables those types to have those instances; we could have added them before, but the instances would have just had to live inside the `generics-rep` library. Also I think we should avoid talking about the "why" in this document.
```suggestion
The `Generic` type class now lives in the `prelude` library. This makes it easier for us to provide `Generic` instances for common types which should have them such as `Tuple` or `Either`.
```

Reply:
This change makes sense and is a bit more readable because there's less content. So :+1: from me.